src/ClientHandle.cpp

```diff
@@ -1120,34 +1116,30 @@ void cClientHandle::HandleLeftClick(int a_BlockX, int a_BlockY, int a_BlockZ, eB
 		When the user breaks a fire block, the client send the wrong block location.
 		We must find the right block with the face direction. */
 
-		int BlockX = a_BlockX;
-		int BlockY = a_BlockY;
-		int BlockZ = a_BlockZ;
-		AddFaceDirection(BlockX, BlockY, BlockZ, a_BlockFace);
+		Vector3i BlockPos = a_BlockPos;
+		AddFaceDirection(BlockPos, a_BlockFace);
 
-		if (cChunkDef::IsValidHeight(BlockY) && cBlockInfo::IsClickedThrough(m_Player->GetWorld()->GetBlock({ BlockX, BlockY, BlockZ })))
+		if (cChunkDef::IsValidHeight(BlockPos.y) && cBlockInfo::IsClickedThrough(m_Player->GetWorld()->GetBlock(BlockPos)))
```

Comment:
Now you added the `ChunkDef::IsValidHeight(Vector3)` function, should we standardise on that form rather than the `Vec.y` form? I don't have any particular preference, and probably has negligible performance impact but it's slightly confusing with two overloaded signatures.